docs/release-notes.rst

```diff
@@ -0,0 +1,16 @@
+Release notes
+=============
+
```

Comment:
Let's seed the release note section for 3.0.1 

```
.. release_3.0.1:

3.0.1 (unreleased)
------------------

...
```

Reply:
We shouldn't have the 3.0.1 title in the 3.0.0 release, so shouldn't we hold off here until after release?

Reply:
We typically seed the release notes with the next release so contributors have a spot to start adding release notes. We can do this right after the 3.0.0 release I suppose.

Reply:
In numcodecs, we also seed a section but without version number. https://github.com/zarr-developers/numcodecs/blob/main/docs/release.rst?plain=1#L16

Reply:
That's after a release is made, the notes for the current release don't have an unreleased section: https://numcodecs.readthedocs.io/en/stable/release.html